src/tests/suite/daos_test_common.c

```diff
@@ -1018,3 +1020,115 @@ get_daos_prop_with_user_acl_perms(uint64_t perms)
 	D_FREE(user);
 	return prop;
 }
+
+/* JSON output handling for dmg command */
+static struct json_object *daos_dmg_json_contents(const char *dmg_cmd)
+{
+	long int  size = 0;
+	char *content = NULL;
+	char *filename = "/tmp/daos_dmg.json";
```

Comment:
Where does this file get cleaned up?

Please take a look at my PR #3140. I suggest that you incorporate that work into this one so that you can dynamically build a dmg commandline and get output back without having to write to a file.

Reply:
fopen(filename, "w+") does the clean.

"w+" | write/update: Create an empty file and open it for update (both for input and output). If a file with the same name already exists its contents are discarded and the file is treated as a new empty file.
-- | --

Reply:
My concern is more that this file will be left over after testing, which is bad form. At a minimum, it seems as if you should `unlink()` it at the end of this function after you've read its contents.

Reply:
Agree, it should be unlinked. Done.